tests/test_cmdline_options.py

```diff
@@ -145,7 +145,7 @@ def cli_arguments_combinations(draw):
     # to point to a parser that we want fawltydeps to use.
     # Following code ensures that we do not encounter trying to match
     # explicit file with incorrect parser (square in the oval-shaped hole problem)
-    drawn_deps = draw(deps_option_strategy(dep_paths))
+    drawn_deps = [] if not dep_paths else draw(deps_option_strategy(dep_paths))
```

Comment:
The idea with the drawing from `deps_option_strategy` with additional `map` function should be enough to handle empty list case. Was there some problem with this parsing? The `code_option_strategy` should be symmetrical.

Reply:
Without this line, this test case fails with:
```
hypothesis.errors.InvalidArgument: Cannot sample from a length-zero sequence.
```
I'm not exactly sure why, but the traceback from Hypothesis does indicate that `draw()` is not happy with being passed an empty sequence. I first thought the following line further down would the also be problematic:
```python
    code = draw(code_option_strategy(code_paths))
```
but it isn't and I guess that's because `code_paths` always contains at least the STDIN option, and can therefore never be empty.

Reply:
I added the fixup to this in 03af92f7960850ecab3e95605c8f8e76ae0676c9. We want to keep the generation logic as close to the strategy definition as possible.

The problem you had looks like the `list` strategy could not use an empty list as an input. That is why I added a guard for an empty input to the `code` and `deps` strategies.